src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,343 @@
+package school.lemon.changerequest.java.io.hw1;
+
+
+import school.lemon.changerequest.java.io.hw1.exception.*;
+
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+
+public class IFileManagerImp implements IFileManager {
+
+
+    File currentDirectory;
+
+    boolean directoriesFirst = true;
+
+
+    public IFileManagerImp(String path) {
+
+        this.currentDirectory = new File(path);
+    }
+
+    public IFileManagerImp() {
+        this.currentDirectory = new File(System.getProperty("user.dir"));
+    }
+
+    @Override
+    public String ls() {
+        return FilesInfo(filter(Order.BY_LEXICOGRAPHICAL_ASCENDING, true, fileFilterForAll()));
+    }
+
+    @Override
+    public String ls(Order order) {
+        return FilesInfo(filter(order, true, fileFilterForAll()));
+    }
+
+    @Override
+    public String ls(Order order, boolean directoriesFirst) {
+
+        return FilesInfo(filter(order, directoriesFirst, fileFilterForAll()));
+    }
+
+    @Override
+    public List<File> filter(Order order, boolean directoriesFirst, FileFilter fileFilter) {
+        File[] paths;
+        paths = currentDirectory().listFiles(fileFilter);
+        List<File> result = new ArrayList<>();
+        if (directoriesFirst == true) {
+            switch (order) {
```

Comment:
You have got lots of duplicated code.
Could you use your `if` and `switch` to decide only what comparator you need and move other operations(as they are actually duplicated) out of `if-else-switch`?